### PR TITLE
RPP test suite upgrade

### PR DESCRIPTION
**Pull request by r-abishek**
- Re-orders and buckets kernels by functionality-group for all tests
- UnitTests dumps output images in respective folders based on the functionality group
- PerformanceTests generates results based on functionality group
- OpenCV output dumps for all UnitTests corrected to show images without padding
- Profile option with rocprof integrated into all test suites
- Update default performance measurement provided by the repository to 8 224x224 images
- Support to test kernels individually or by range
- Clean up unwanted or commented code
- Modify tabs to spaces and formatting

### CODE REVIEW

File: utilities/rpp-performancetests/OCL_NEW/BatchPD_ocl_pkd3.cpp [hunk 1]
```diff
@@ -1893,92 +2099,146 @@ int main(int argc, char **argv)
...
+            Rpp32u crop_pos_y[images];
+            Rpp32f mean[images];
+            Rpp32f stdDev[images];
+            Rpp32u mirrorFlag[images];
             for (i = 0; i < images; i++)
             {
-                kernelSize[i] = 5;
+                dstSize[i].height = 100;
+                dstSize[i].width = 100;
+                if (maxDstHeight < dstSize[i].height)
```
Comment (rrawther) [inline utilities/rpp-performancetests/OCL_NEW/BatchPD_ocl_pkd3.cpp:2160]: betterr to call std::min and std::max for these comparisons

File: utilities/rpp-performancetests/OCL_NEW/BatchPD_ocl_pkd3.cpp [hunk 2]
```diff
@@ -3369,6 +3312,41 @@ int main(int argc, char **argv)
 
             break;
         }
+        case 75:
+        {
+            test_case_name = "fast_corner_detector";
+            missingFuncFlag = 1;
```
Comment (rrawther) [inline utilities/rpp-performancetests/OCL_NEW/BatchPD_ocl_pkd3.cpp:3318]: There are lots of missing functions flag. Are those expected? 
We need to print out error code saying not implemented for those cases

**Review by rrawther — APPROVED** [review]
@paveltc to run all unit-tests before merging this PR
@r-abishek please make the reviewed changes

**Comment by paveltc** [comment]
@rrawther and @r-abishek I ran all of the tests and it seems that all of them run fine with these changes. Everything is very well organized now.

**Pull request merged**